controllers/class.tapestry-controller.php

```diff
@@ -0,0 +1,337 @@
+<?php
+require_once dirname(__FILE__) . "/../utilities/class.tapestry-errors.php";
+require_once dirname(__FILE__) . "/../utilities/class.tapestry-helpers.php";
+require_once dirname(__FILE__) . "/../utilities/class.tapestry-user-roles.php";
+require_once dirname(__FILE__) . "/../utilities/class.tapestry-node-permissions.php";
+require_once dirname(__FILE__) . "/../interfaces/interface.tapestry-controller.php";
+
+/**
+ * Add/update/retrieve a Tapestry
+ * 
+ */
+class TapestryController implements ITapestryController
+{
+    private $postId;
+
```

Comment:
Missing other private properties for nodes, links, groups, and settings (including rootId).